### ISSUE TITLE
Offer more replacements than just the package's version

### ISSUE THREAD

**Issue by Vogtinator** [openSUSE/obs-service-replace_using_package_version#7]
While it obviously goes against the package's name, it would be useful to have the package's `%{description}`, `%{summary}`, `%{license}`, `%{url}`, etc. available as replacements as well.

**Comment by davidcassany** [comment]
You mean grabbing this values form an rpm metadata and use them as values for replacement? I guess it should be possible.

Probably sorting out a better name (and see the pros and cons of the change) is something that could be discussed after a round of new features.

**Comment by Vogtinator** [comment]
Yup, this would make #4 more relevant.

**Comment by akshat5302** [comment]
I would like to contribute to it could you just elaborate the issue @Vogtinator @davidcassany

**Comment by dcermak** [comment]
@akshat5302 At the moment the service offers to find the version of a package in the buildroot and insert it in the build recipe file. For example, given this `_service` file:
```xml
<service name="replace_using_package_version" mode="buildtime">
  <param name="regex">%%TAG%%</param>
  <param name="package">mariadb</param>
</service>
```
the service would then look for the package `mariadb`, fetch its version, look for the build recipe in this package (e.g. if this was a rpm package, that would be the `spec` file) and replace `%%TAG%%` in the build recipe with the version of `mariadb`.

Now this feature request is about allowing to extract additional information from the `mariadb` package, not only the version, but also other metadata of the rpm package, e.g. the description.

You generally get these metadata via rpm:
```ShellSession
f4f0280b5e5f:/ # rpm -q --qf "%{DESCRIPTION}" mariadb
MariaDB is an open-source, multi-threaded, relational database management
system. It's a backward compatible, drop-in replacement branch of the
MySQL Community Server.

This package only contains the server-side programs.
```

To get this in, you'd have to add a new command line flag to [`replace_using_package_version.py`](https://github.com/openSUSE/obs-service-replace_using_package_version/blob/master/replace_using_package_version/replace_using_package_version.py) that would allow to specify which metadata to extract.

Does that help you out a bit? If anything is unclear, please don't hesitate to reach out.

**Comment by dcermak** [comment]
No, I wouldn't create a separate function for each metadata that we want to fetch, as that will be essentially the same call every time. The only difference will be the string after `--queryformat`.

**Comment by akshat5302** [comment]
And that string we can take as input from user and put that in `%{}` block  which kind of metadata they want to see from rpm package I think so

**Comment by dcermak** [comment]
Yes, but we really should limit the allowed input to a few sensible options (e.g. `version`, `description`, `epoch`), as the user shouldn't be allowed to inject arbitrary strings here (that _could_ even lead to vulnerabilities).

**Comment by dcermak** [comment]
If you want to work on this simply open a pull request, assigning the issue to you will not change a thing ;-)

**Comment by akshat5302** [comment]
Yes sure 😊

**Comment by dcermak** [comment]
@akshat5302 In case you are still interested in working on this issue and are stuck or want help, feel free to reach out for guidance. No pressure from my side though, feel free to proceed at your own pace.

**Comment by dcermak** [comment]
No worries @akshat5302, I was just asking whether you want help with anything. This is really not time critical and there's no pressure to get this done by $date.

**Comment by akshat5302** [comment]
Let me first read it till tomorrow if any problem occurs will ask you for help thanks for your concern 👍🏽

**Comment by akshat5302** [comment]
hey @dcermak could you please help me out in filling proposal for gsoc2022 for opensuse

**Comment by dcermak** [comment]
> hey @dcermak could you please help me out in filling proposal for gsoc2022 for opensuse

Sure, if you need any help feel free to ping me via email or IM.

**Comment by akshat5302** [comment]
Hey, @dcermak I'm again up to solve this issue. Tell me where to start or give a brief overview of this issue. Sorry for this much amount of delay

**Comment by dcermak** [comment]
Well, I now have to apologize for this huge delay (this fell into my backlog when I was on vacation…). Are you still interested in working on this and need help @akshat5302?

**Comment by akshat5302** [comment]
Yes @dcermak

**Comment by dcermak** [comment]
How can I help you?
